src/jaxsim/physics/algos/aba.py

```diff
@@ -81,27 +93,36 @@ def aba(
         jtp.MatrixJax,
         jtp.MatrixJax,
         jtp.MatrixJax,
+        jtp.MatrixJax,
+        jtp.MatrixJax,
+        jtp.MatrixJax,
     ]
 
-    pass_1_carry = (i_X_λi, v, c, MA, pA, i_X_0)
+    pass_1_carry = (i_X_λi, v, c, m_v, m_c, MA, pA, pR, i_X_0)
 
     # Pass 1
     def loop_body_pass1(carry: Pass1Carry, i: jtp.Int) -> Tuple[Pass1Carry, None]:
         ii = i - 1
-        i_X_λi, v, c, MA, pA, i_X_0 = carry
+        i_X_λi, v, c, m_v, m_c, MA, pA, pR, i_X_0 = carry
 
         # Compute parent-to-child transform
         i_X_λi_i = i_X_pre[i] @ pre_X_λi[i]
         i_X_λi = i_X_λi.at[i].set(i_X_λi_i)
 
         # Propagate link velocity
-        vJ = S[i] * qd[ii] if qd.size != 0 else S[i] * 0
+        vJ = S[i] * qd[ii] * (qd.size != 0)
+        m_vJ = m_S[i] * qd[ii] * (qd.size != 0)
```

Comment:
Did you validate jit compilation? I remember I had to use the if/else statement in this way otherwise jax was complaining

Reply:
Yes, it did not complain, but we can further investigate on this